File: DesktopCore/src/burlton/desktopcore/code/bean/HyperlinkAdaptor.java
package burlton.desktopcore.code.bean;

import java.awt.Component;
import java.awt.Cursor;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

import burlton.core.code.util.Debug;

public class HyperlinkAdaptor extends MouseAdapter
{
	private HyperlinkListener listener = null;
	private Component listenerWindow = null;
	
	public HyperlinkAdaptor(HyperlinkListener listener)
	{
		if (!(listener instanceof Component))
		{
			Debug.stackTrace("Creating HyperlinkAdaptor with non-component: " + listener);
		}
		
		this.listener = listener;
		this.listenerWindow = (Component)listener;
	}
	
	@Override
	public void mouseClicked(MouseEvent arg0)
	{
		listener.linkClicked(arg0);
	}

	@Override
	public void mouseMoved(MouseEvent arg0)
	{
		if (listener.isOverHyperlink(arg0))
		{
			listenerWindow.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
		}
		else
		{
			listenerWindow.setCursor(Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR));
		}
	}

	@Override
	public void mouseExited(MouseEvent arg0)
	{
		listenerWindow.setCursor(Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR));
	}
}

File: src/main/kotlin/dartzee/core/bean/HyperlinkAdaptor.kt
package dartzee.core.bean

import java.awt.Component
import java.awt.Cursor
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent

class HyperlinkAdaptor(private val listener: IHyperlinkListener) : MouseAdapter()
{
    private val listenerWindow = listener as Component

    override fun mouseReleased(arg0: MouseEvent) = listener.linkClicked(arg0)

    override fun mouseMoved(arg0: MouseEvent)
    {
        if (listener.isOverHyperlink(arg0))
        {
            listenerWindow.cursor = Cursor.getPredefinedCursor(Cursor.HAND_CURSOR)
        }
        else
        {
            listenerWindow.cursor = Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR)
        }
    }

    override fun mouseExited(arg0: MouseEvent?)
    {
        listenerWindow.cursor = Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR)
    }
}
